lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_object.rb

```diff
@@ -161,11 +161,10 @@ def fetch_class(fqname = @class_fqname)
       Benchmark.current_realtime[:fetch_class_count] += 1
       Benchmark.realtime_block(:fetch_class_time) do
         @workspace.datastore(fqname.downcase.to_sym, :class) do
-          ns, name = MiqAeClass.parse_fqname(fqname)
-          ns = fetch_namespace(ns)
-          return nil if ns.nil?
-
-          ns.ae_classes.detect { |c| name.casecmp(c.name).zero? }
+          fqname = fqname[1..-1] if fqname[0] == '/'
+          dname, *partial = fqname.downcase.split('/')
+          domain_id = MiqAeDomain.unscoped.where(MiqAeDomain.arel_table[:name].lower.eq(dname)).where(:relative_path => nil).select(:id)
+          MiqAeClass.where(MiqAeClass.arel_table[:relative_path].lower.eq(partial.join("/"))).find_by(:domain_id => domain_id)
```

Comment:
These lines look similar to the ones in the other PR [here](https://github.com/ManageIQ/manageiq/pull/20050/files#diff-3c054144b5ee9a1945d682e5833f2886R37-R45).  Does it make sense to use that method, or clean it up so the code lives in MiqAeClass or MiqAeDomain itself?